apps/flame-defi/app/swap/hooks/use-swap-button.tsx

```diff
@@ -41,37 +40,29 @@ interface ErrorWithMessage {
 
 const useCheckTokenApproval = (
   topToken: TokenInputState,
-  bottomToken: TokenInputState,
-): EvmCurrency | null => {
+): TokenInputState | null => {
   const { tokenAllowances } = useEvmWallet();
   const topTokenValueBig = tokenStateToBig(topToken);
-  const bottomTokenValueBig = tokenStateToBig(bottomToken);
 
-  const findApproval = (token: TokenInputState) =>
+  const findTokenToApprove = (token: TokenInputState) =>
     tokenAllowances.find(
       (allowanceToken) => allowanceToken.symbol === token.token?.coinDenom,
     );
 
-  const topTokenNeedingApproval = topToken.token
-    ? findApproval(topToken)
-    : null;
-  const bottomTokenNeedingApproval = bottomToken.token
-    ? findApproval(bottomToken)
+  const topTokenNeedingApproval = topToken?.token
```

Comment:
You do not need the optional chain operator here! `topToken` will NEVER be null or undefined.